src/CarbonAware/src/Model/EmissionsForecast.cs

```diff
@@ -36,4 +38,49 @@ public record EmissionsForecast
     /// Gets or sets the optimal data point within the ForecastData set.
     /// </summary>
     public EmissionsData OptimalDataPoint { get; set; }
+
+
+    public void Validate()
+    {
+        var errors = new Dictionary<string, List<string>>();
+        var firstDataPoint = ForecastData.First();
+        var lastDataPoint = ForecastData.Last();
+        var minTime = firstDataPoint.Time;
+        var maxTime = lastDataPoint.Time + lastDataPoint.Duration;
+
+        if (StartTime >= EndTime)
+        {
+            AddErrorMessage(errors, "startTime", "startTime must be earlier than endTime");
+        }
+
+        if (StartTime < minTime || StartTime > maxTime)
+        {
+            AddErrorMessage(errors, "startTime", $"startTime must be within time range of the forecasted data, '{minTime}' through '{maxTime}'");
+        }
+
+        if (EndTime < minTime || EndTime > maxTime)
+        {
+            AddErrorMessage(errors, "endTime", $"endTime must be within time range of the forecasted data, '{minTime}' through '{maxTime}'");
+        }
+
+        if (errors.Keys.Count > 0)
+        {
+            ArgumentException error = new ArgumentException("Invalid EmissionsForecast");
+            foreach (KeyValuePair<string, List<string>> message in errors)
+            {
+                Console.WriteLine($"Pair here: {message.Key}, {message.Value}");
```

Comment:
nit-Q, could this use the `_logger.LogError` instead of `Console.WriteLine` ? -- which I realize might be overkill (due to amount of changes required)...

Reply:
line removed